### PR TITLE
V0.5 fixes

### PR DESCRIPTION
**Pull request by cliffckerr**
### Description
Various bugfixes; namely:
- #387
- #437
- #479
- #489
- #490
- #495
- #506
- #512
- #513
- #519

### Checklist
- [x] Code commented & docstrings added
- [x] New tests were needed and have been added
- [x] A new version number was needed & changelog has been updated
- [x] A new PyPI version needs to be released

### CODE REVIEW

File: starsim/distributions.py
Comment (kaa2102) [inline starsim/distributions.py:1]: The move from 'delta' to 'constant' is just a lot more logical and clear to understand for users.

File: tests/test_networks.py
Comment (kaa2102) [inline tests/test_networks.py:1]: Adding StaticNet to tests expands test coverage! DynamicNet is tested via MaternalNet and RandomNet. This leaves: NullNet and SexualNet as classes excluded from testing in 'network.py'. I will add a test for this for Issue #515.

**Review by kaa2102 — APPROVED** [review]
Straightforward changes. Testing passed and I didn't see any errors or issues so this PR is good to go from my perspective.

File: starsim/modules.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def check_requires(self, sim):
             raise Exception(errormsg)
         return
 
-    def initialize(self, sim):
+    def init_pre(self, sim):
```
Comment (robynstuart) [inline starsim/modules.py:107]: i reckon there should be a comment somewhere that attempts to clarify for users which things should go in __init__ vs init_pre vs init_post. maybe in these docstrings?

File: starsim/distributions.py [hunk 2]
```diff
@@ -587,7 +587,7 @@ def plot_hist(self, n=1000, bins=None, fig_kw=None, hist_kw=None):
 
 # Add common distributions so they can be imported directly; assigned to a variable since used in help messages
 dist_list = ['random', 'uniform', 'normal', 'lognorm_ex', 'lognorm_im', 'expon',
-             'poisson', 'weibull', 'delta', 'randint', 'bernoulli', 'choice']
+             'poisson', 'weibull', 'const', 'randint', 'bernoulli', 'choice']
```
Comment (robynstuart) [inline starsim/distributions.py:590]: not `constant`? it's only 8 letters, shorter than the `lognorm_**` options! at least make an alias?

**Review by robynstuart — APPROVED** [review]
Looks good, just a couple of small comments

**Pull request merged**